pkg/controller/tas/controllers.go

```diff
@@ -38,5 +43,57 @@ func SetupControllers(mgr ctrl.Manager, queues *queue.Manager, cache *cache.Cach
 	if ctrlName, err := topologyUngater.setupWithManager(mgr, cfg); err != nil {
 		return ctrlName, err
 	}
+
+	// Set up the TAS flavor cache in the background. This is because mgr is configured to list from a cache
+	// and the cache is not populated until the controller is started.
+	go func() {
+		err := syncTasCache(mgr, cache)
```

Comment:
I'm not clear why we need this code if we are already waiting inside scheduler.go?

Reply:
I would probably prefer to have a function like `WaitForTASCache` exposed too which awaits for the caches, by analogy to `mgr.GetCache().WaitForCacheSync(ctx)`. I think it makes sense to create the helper inside the cache.